src/main/kotlin/org/opensearch/observability/index/ObservabilityMetricsIndex.kt

```diff
@@ -0,0 +1,170 @@
+    private const val METRICS_MAPPING_TEMPLATE_NAME = "sso_metric_template"
+    private const val METRICS_MAPPING_TEMPLATE_FILE = "metrics-mapping-template.json"
+    private const val METRICS_INDEX_NAME = "sso_metrics-default-namespace"
+    private const val METRIC_PATTERN_NAME = "sso_metrics-*-*"
+
+    private lateinit var client: Client
+    private lateinit var clusterService: ClusterService
+
+    /**
+     * Initialize the class
+     * @param client The OpenSearch client
+     * @param clusterService The OpenSearch cluster service
+     */
+    fun initialize(client: Client, clusterService: ClusterService): ObservabilityMetricsIndex {
+        this.client = SecureIndexClient(client)
+        this.clusterService = clusterService
+        return this
+    }
+
+    /**
+     * once lifecycle indicate start has occurred - instantiating the mapping template and default data stream
+     */
+    override fun afterStart() {
+        // create default index
+        createDataStream()
+    }
+
+    /**
+     * Create index expecting the index template already exists
+     */
+    @Suppress("TooGenericExceptionCaught")
+    private fun createDataStream() {
+        log.info("$LOG_PREFIX:createDataStream $METRICS_INDEX_NAME API called")
+        if (!isTemplateExists(METRICS_MAPPING_TEMPLATE_NAME)) {
+            createMappingTemplate()
+        }
+        if (!isDataStreamExists(METRICS_INDEX_NAME)) {
+            val request = CreateDataStreamAction.Request(METRICS_INDEX_NAME)
+            try {
+                val actionFuture = client.admin().indices().createDataStream(request)
+                val response = actionFuture.actionGet(PluginSettings.operationTimeoutMs)
+                if (response.isAcknowledged) {
+                    log.info("$LOG_PREFIX:DataStream $METRICS_INDEX_NAME creation Acknowledged")
+                } else {
+                    error("$LOG_PREFIX:DataStream $METRICS_INDEX_NAME creation not Acknowledged")
+                }
+            } catch (exception: ResourceAlreadyExistsException) {
+                log.warn("message: ${exception.message}")
+            } catch (exception: Exception) {
+                if (exception.cause !is ResourceAlreadyExistsException) {
+                    throw exception
+                }
+            }
+        }
+    }
+
+    /**
+     * Create index using the pre-defined mapping template
+     */
+    @Suppress("TooGenericExceptionCaught", "MagicNumber")
+    private fun createMappingTemplate() {
+        log.info("$LOG_PREFIX:createMappingTemplate $METRICS_INDEX_NAME API called")
+        if (!isTemplateExists(METRICS_MAPPING_TEMPLATE_NAME)) {
+            val classLoader = ObservabilityMetricsIndex::class.java.classLoader
+            val indexMappingSource = classLoader.getResource(METRICS_MAPPING_TEMPLATE_FILE)?.readText()!!
+            val settings = Settings.builder()
+                .put("index.number_of_shards", 3)
```

Comment:
why choose 3? can user change this?

Reply:
should we load this from an Environment variable ? what is the best practice here ?

Reply:
I've removed the eagerly creation of a default index on plugin loading, this should be done by user after the Observability plugin is loaded